php_phongo.c

```diff
@@ -3962,11 +3957,35 @@ PHP_RSHUTDOWN_FUNCTION(mongodb)
 /* {{{ PHP_GSHUTDOWN_FUNCTION */
 PHP_GSHUTDOWN_FUNCTION(mongodb)
 {
+	php_phongo_pclient_t* pclient;
+
+	/* Destroy mongoc_client_t objects in reverse order. This is necessary to
+	 * prevent segmentation faults as clients may reference other clients in
+	 * encryption settings. */
+	ZEND_HASH_REVERSE_FOREACH_PTR(&mongodb_globals->persistent_clients, pclient)
+	{
+		php_phongo_pclient_destroy(pclient);
+	}
+	ZEND_HASH_FOREACH_END();
+
+	/* Destroy HashTable for persistent clients. mongoc_client_t objects have
+	 * already been destroyed. */
+	zend_hash_destroy(&mongodb_globals->persistent_clients);
+
```

Comment:
This entire block will be refactored by [PHPC-1737](https://jira.mongodb.org/browse/PHPC-1737) in a subsequent PR, unless you'd prefer I lump it in to this one.

Reply:
Subsequent PR sounds good, no need to lump it all into one PR.